deta/base.py

```diff
@@ -46,63 +51,201 @@ def append(self, value: typing.Union[dict, list, str, int, float, bool]):
-        )
+        if isinstance(data, dict):
+            self.client.request(
+                method,
+                url,
+                headers=headers,
+                body=json.dumps(data),
+            )
+        else:
+            self.client.request(
+                method,
+                url,
+                headers=headers,
+                body=data,
+            )
         res = self.client.getresponse()
         status = res.status
         payload = res.read()
+        
+        if status in [200, 202, 201, 207, 404]:
+            if status == 404:
+                return status, None
+            if (no_read):
+                return status, res
+            try:
+                payload = json.loads(payload)
+            except:
+                pass
+            return status, payload
+
+        raise urllib.error.HTTPError(
+            url, status, res.reason, res.headers, res.fp)
+
+class Drive(_Service):
+    def __init__(self, drive_name:str=None, project_key:str=None, project_id:str=None, host:str=None):
+        super().__init__(project_key=project_key, project_id=project_id, host=host,
+                         name=drive_name)
+        assert drive_name, "Please provide a Drive name. E.g 'mydrive"
+
+    def get(self, name:str=None) -> typing.Optional[BufferedIOBase]:
+        assert name, "Please provide the name of the file to get."
+        code,res = self._request(f"/files/download?name={name}", "GET", no_read=True)
+        stream = BytesIO()
+        if isinstance(res, bytes):
+            stream.write(res)
+            stream.seek(0)
+            return stream
+        raise Exception("Response was not of type bytes; got "+str(type(res))+" instead.")
+        
+
+    def deleteMany(self, names:typing.List[str]) -> typing.Optional[dict]:
+        assert names, "Names is empty"
+        _, res = self._request("/files", "DELETE", {"names": names})
+        if isinstance(res, dict):
+            return res
+        raise Exception("Result not a dict; got "+str(type(res))+" instead.")
+
+    def delete(self, name: str) -> typing.Optional[str]:
+        """Delete an item from drive
+        name: the name of item to be deleted
+        """
+        if name == "":
+            raise ValueError("Name is empty")
+        # encode key
+        name = quote(name, safe="")
+        status, payload = self._request("/files", "DELETE", {"names":[name]})
+        if (status == 404):
+            return None
+        if (isinstance(payload, dict)):
+            if len(payload["failed"]) > 0:
+                raise Exception(f"Deletion failed for: {payload['failed']}")
+            return payload["deleted"]
+        raise Exception("Payload was not a dict; got "+str(type(payload))+" instead.")
+
+    def put(self, name:str, data:typing.Union[str, BufferedIOBase, bytes]=None, *, path:str=None, content_type:str=None) -> str:
+        chunk_size = 104857600  # TODO 100MB threshold needs tuning
```

Comment:
can you please define it as a constant at the top of the file like 
`CHUNK_SIZE`